launcher/modplatform/ModIndex.h

```diff
@@ -30,13 +30,21 @@ class QIODevice;
 
 namespace ModPlatform {
 
-enum ModLoaderType { NeoForge = 1 << 0, Forge = 1 << 1, Cauldron = 1 << 2, LiteLoader = 1 << 3, Fabric = 1 << 4, Quilt = 1 << 5 };
+enum ModLoaderType {
+    NeoForge = 1 << 0,
+    Forge = 1 << 1,
+    Cauldron = 1 << 2,
+    LiteLoader = 1 << 3,
+    Fabric = 1 << 4,
+    Quilt = 1 << 5,
+    DataPack = 1 << 6
```

Comment:
Can't we just detect in ModrinthAPI, where the search string is built that the resource type is datapack and replace the loader bit with a custom one without needing a fake loader?

Reply:
ResourceType is not in SearchArgs. I thought it would be a bit clunky to have a dataPacks boolean in SearchArgs too, and adding ResourceType wouldn't make sense either as it would only be used on Modrinth for data packs

Reply:
Also I don't think it is that illogical - Modrinth considers it a mod loader and it basically is - either implemented by vanilla for each world or by a global pack loader for all worlds.

Reply:
Wait got a bit confused - yes it is in SearchArgs but not in VersionSearchArgs. Since VersionSearchArgs contains an IndexedPack maybe the ResourceType could be put in there?

Reply:
The trouble is that would work for now, but it wouldn't work if we want to support downloading datapacks to the mods folder (for the case of "datapacks" which support mod loaders)

Reply:
Sure, we can put the resorceType in the IndexedPack or the VersionSearchArgs anything that would allow us not to propagate DataPack as a loader.

Reply:
Can you look into CF implementation also and see how it handles datapacks?
Maybe there will be a better way.

Reply:
>  Sure, we can put the resorceType in the IndexedPack or the VersionSearchArgs anything that would allow us not to propagate DataPack as a loader.

That would kind of conflict with allowing installing datapack jars to the mod folder. (Modrinth allows datapacks to contain versions repackages as jars for forge and fabric etc.)

Reply:
NVM i  think that makes the most sense in the long term if we actually want maintainable code :trollface:

Reply:
Turned out to be too hard
We also have other loader types for cauldron (which is literally for servers) so shrug